### ISSUE TITLE
実装: non-playing AI の短時間学習・ONNX export・評価を1コマンドで実行するCLI

### ISSUE THREAD

**Issue by eletim** [eletim/napoleon-web#221]
Parent: #202
Investigation: #201
Base branch: `feature/issue-202-non-playing-ai`

## 背景

Issue #202 配下で bidding / adjutant / exchange の各RL rollout、PPO trainer、ONNX export、runtime integration、evaluation までは実装済み。

ただし現状は、個々のAPI/CLIやテストを組み合わせれば縦通し確認できるものの、ユーザーが親ブランチ上で実際に短時間学習を回し、生成artifactを使って評価するまでの手順が分散している。

mainへ統合する前の実動作確認をしやすくするため、non-playing AIの短時間学習・export・評価を1コマンドで実行できるCLIを追加する。

## 目的

1コマンドで少なくとも以下を順番に実行できるようにする。

1. bidding RL rollout dataset生成
2. bidding PPO学習
3. bidding ONNX export
4. adjutant RL rollout dataset生成
5. adjutant PPO学習
6. adjutant ONNX export
7. exchange RL rollout dataset生成
8. exchange 3-step PPO学習
9. exchange ONNX export
10. 生成した3つのnon-playing artifact + frozen playing policyを組み合わせたfull-policy評価
11. 実行結果・artifact path・主要評価指標をまとめて表示 / JSON保存

## 初版の用途

本番長時間学習ではなく、まずは親ブランチの実動作確認と短い実験を主目的とする。

デフォルトは小規模にし、ユーザーがgames / epochs / batch size / seed / output directory等を引数で増やせるようにする。

## 方針

- 既存の rollout / trainer / ONNX export / evaluation 実装を再利用し、学習ロジックを重複実装しない
- playing policyは repo管理済み `benchmarks/playing-policies/ppo-separated-v1000` をデフォルトのfrozen policyとする
- bidding / adjutant / exchange は各phaseの既存PPO実装を利用する
- exchangeは `sequential-card-v1`
- 途中成果物をoutput directory配下にphase別で保存する
- 失敗したstageが分かるようにfail-fastし、stage名と原因を表示する
- 再実行時に既存outputを誤って上書きしない。明示的なoverwrite/resume相当が必要なら設計する
- CPUで動作確認可能にする。device指定が既存CLIで可能なら統合CLIから透過的に渡せるようにする

## 想定CLI

名前は既存命名規則に合わせて決めること。例:

`napoleon-run-non-playing-rl`

想定例:

```bash
cd python
uv run napoleon-run-non-playing-rl \
  --output-dir /tmp/napoleon-non-playing-smoke \
  --games 20 \
  --epochs 1 \
  --batch-size 32 \
  --seed 202
```

実際の引数体系は既存実装との整合性を優先してよい。

## 出力ディレクトリ例

```text
<output-dir>/
  bidding/
    dataset/
    checkpoint.pt
    policy.onnx
    policy.json
  adjutant/
    dataset/
    checkpoint.pt
    policy.onnx
    policy.json
  exchange/
    dataset/
    checkpoint.pt
    policy.onnx
    policy.json
  evaluation.json
  run-summary.json
```

既存artifact契約に合わせて名称変更してよい。

## 評価

最低限、生成したfull non-playing policyで固定seedゲームを複数完走させる。

run summary / evaluationでは可能な範囲で以下を記録する。

- games / completed / failed
- phase別 ONNX decision count
- fallback count
- illegal action count
- bidding pass率
- bid 13〜19分布
- suit分布
- 平均宣言枚数
- 宣言成功率
- role別平均reward
- adjutant選択分布
- exchange選択傾向
- 各artifact path / metadata path
- seed / training settings

既存evaluation APIで取得可能な項目を優先し、不足分のためだけに大規模な評価基盤再設計はしない。

## Acceptance Criteria

- 1コマンドで bidding → adjutant → exchange の rollout / PPO train / ONNX export と full-policy evaluation まで完走できる
- デフォルトまたはsmoke設定で短時間に実行可能
- 各phaseで生成されたcheckpoint / ONNX / metadataのpathが明確に出力される
- 生成ONNXを実際のruntime loaderで読み込んで評価する
- full-policy fixed-seed evaluationが完走する
- fallback count / illegal action countを確認できる
- run summaryをJSONで保存する
- 同じseed / settingsなら再現可能な範囲でdeterministicである
- 途中stage失敗時に、どのstageで何が失敗したか明確なエラーを出す
- 既存の個別CLI、training API、artifact互換性を壊さない
- CLIの使用例をREADME等に記載する
- 小規模end-to-end smoke testを追加する

## 非目標

- 長時間学習向けの高度なresume orchestration
- bidding / adjutant / exchangeのjoint neural network化
- reward値の調整
- PPOアルゴリズム変更
- playing policy再学習
- C++ backendへの全面移植

## PR運用

このIssueは #202 の追加子Issueとして扱う。
専用ブランチを最新の `feature/issue-202-non-playing-ai` から作り、PRのbaseも `feature/issue-202-non-playing-ai` とする。
親PR #203 はmainへまだMergeしない。
